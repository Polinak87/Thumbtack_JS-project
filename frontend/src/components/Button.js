import React from 'react';

export default class Button extends React.Component {
  constructor(props) {
    super(props);
    this.style = this.style.bind(this);
  }

  style(type){
    if (type=='Add to market') {
    return "button is-block is-success is-large is-fullwidth";
    };
    if (type=='Remove from market'){
    return "button is-block is-danger is-large is-fullwidth";
    };
    if (type=='Complete application'){
      return "button is-block is-success is-large is-fullwidth";
    };
    if (type=='Reject application'){
      return "button is-block is-danger is-large is-fullwidth";
    };
    if (type=='Cancel application'){
      return "button is-block is-danger is-large is-fullwidth";
    };
  };

  render() {
    let {type, onClick, id} = this.props;
    console.log(type);
    return (
      <>
        <button className={this.style(type)} onClick={() => onClick(id, type)}> {type} </button>
      </>
    );
  }
}

