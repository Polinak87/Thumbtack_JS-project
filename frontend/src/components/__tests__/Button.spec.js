import React from 'react';
import { render, fireEvent, cleanup } from 'react-testing-library';
import Button, { green, large } from '../Button';

afterEach(cleanup);

describe('Button', () => {
  it('has button name', () => {
    const text = 'Add to market';
    const props = {
      className: `${large} ${green}`,
      id: '1',
      userId: undefined,
      onClick: jest.fn(),
    }
    const { getByText } = render(<Button {...props}>{text}</Button>);
    expect(getByText(text).textContent).toBe(text);
  });

  it('clicks button', () => {
    const text = 'Add to market';
    const id = '1';
    const userId = undefined;
    const onClick = jest.fn();
    const props = {
      className: `${large} ${green}`,
      id,
      userId,
      onClick,
    }
    const { getByText } = render(<Button {...props}>{text}</Button>);
    fireEvent.click(getByText(text));
    expect(onClick).toBeCalledWith(id, userId);
  });

  it('renders correctly', () => {
    const text = 'Add to market';
    const props = {
      className: `${large} ${green}`,
      id: '1',
      userId: undefined,
      onClick: jest.fn(),
    }
    const { container } = render(<Button {...props}>{text}</Button>);
    expect(container.firstChild).toMatchSnapshot();
  });
});