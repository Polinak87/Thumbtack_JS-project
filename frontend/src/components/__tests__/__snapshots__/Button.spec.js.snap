// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<button
  class="button is-block is-large is-fullwidth is-success"
>
  Add to market
</button>
`;
